libbeat/esleg/eslegclient/connection.go

```diff
@@ -110,6 +110,9 @@ func NewConnection(s ConnectionSettings) (*Connection, error) {
 	if st := s.Observer; st != nil {
 		dialer = transport.StatsDialer(dialer, st)
 		tlsDialer = transport.StatsDialer(tlsDialer, st)
+		logger := logp.NewLogger("esclientleg")
+		dialer = transport.LoggingDialer(dialer, logger)
+		tlsDialer = transport.LoggingDialer(tlsDialer, logger)
```

Comment:
We always want the logging. Lets move the `LoggingDialer` out of the if statement.